### PR TITLE
Tganley/bugfix/adcs service testing

### PR DESCRIPTION
**Pull request by tganley**
All subservices required by the ADCS team are working

### CODE REVIEW

File: ex2_hal/adcs/equipment_handler/src/adcs_handler.c [hunk 1]
```diff
@@ -1306,10 +1310,12 @@ ADCS_returnState ADCS_get_bootloader_state(uint16_t *uptime, uint8_t *flags_arr)
     state = adcs_telemetry(GET_BOOTLOADER_STATE_ID, telemetry, 6);
     *uptime = (telemetry[1] << 8) + telemetry[0];
 
-    for (int k = 0; k < 2; k++) {
-        for (int i = 0; i < 8; i++) {
-            *(flags_arr + k * 8 + i) = (telemetry[2 + k] >> i) & 1;
-        }
+
+    for (int i = 0; i < 8; i++) {
```
Comment (Taylorr82) [inline ex2_hal/adcs/equipment_handler/src/adcs_handler.c:1314]: Could you add a comment explaining this? For future maintenance

File: ex2_hal/adcs/equipment_handler/src/adcs_handler.c [hunk 2]
```diff
@@ -3018,25 +3042,38 @@ ADCS_returnState ADCS_set_mtm_config(mtm_config params, uint8_t mtm) {
...
-    raw_val_offset.x = params.channel_offset.x / coef;
-    raw_val_offset.y = params.channel_offset.y / coef;
-    raw_val_offset.z = params.channel_offset.z / coef;
-    memcpy(&command[7], &raw_val_offset, 6);
+    float coef = 100;
+    raw_val_angle.x = params.mounting_angle.x * coef;
+    raw_val_angle.y = params.mounting_angle.y * coef;
+    raw_val_angle.z = params.mounting_angle.z * coef;
+    command[1] = raw_val_angle.x & 0xFF;
+    command[2] = raw_val_angle.x >> 8;
```
Comment (Taylorr82) [inline ex2_hal/adcs/equipment_handler/src/adcs_handler.c:3050]: When doing bit shfits like this it's common to and them with FF after shifting or with FF00 before shifting

File: ex2_hal/adcs/equipment_handler/src/adcs_handler.c [hunk 3]
```diff
@@ -3018,25 +3042,38 @@ ADCS_returnState ADCS_set_mtm_config(mtm_config params, uint8_t mtm) {
...
     for (int i = 0; i < 3; i++) {
-        cell[i] = params.sensitivity_mat[4 * i] / coef; // diagonal
+        cell[i] = params.sensitivity_mat[4 * i] * coef; // diagonal
+        cell[3 + i] = params.sensitivity_mat[1 + i] * coef;
+        cell[6 + i] = params.sensitivity_mat[5 + i] * coef;
     }
-    for (int i = 0; i < 3; i++) {
-        cell[3 + i] = params.sensitivity_mat[1 + i] / coef;
-        cell[6 + i] = params.sensitivity_mat[5 + i] / coef;
+    for(int i = 0; i < 9; i++){
```
Comment (Taylorr82) [inline ex2_hal/adcs/equipment_handler/src/adcs_handler.c:3073]: What's going on here? This PR needs lots of comments explaining the operations

File: ex2_services/Services/source/adcs/adcs_service.c [hunk 4]
```diff
@@ -1251,9 +1260,109 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
...
     case ADCS_SET_CUBESENSE_CONFIG: {
-        cubesense_config config;
-        memcpy(&config, &packet->data[IN_DATA_BYTE], sizeof(cubesense_config));
-        status = ADCS_set_cubesense_config(config);
+        cubesense_config cs_config;
+        char file_name[30];
+        memcpy(file_name, &packet->data[IN_DATA_BYTE], 30);
+
+        strncat(file_name, ".txt", 4);
+        int32_t iErr = red_chdir("VOL0:/adcs");
```
Comment (Taylorr82) [inline ex2_services/Services/source/adcs/adcs_service.c:1268]: missing check for iErr

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: ex2_services/Services/source/adcs/adcs_service.c [hunk 5]
```diff
@@ -1251,9 +1260,112 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
...
+        if(iErr == -1){
+            sys_log(ERROR, "Error %d trying to change into adcs directory\r\n", red_errno);
+            status = ADCS_FILESYSTEM_FAIL;
```
Comment (Taylorr82) [inline ex2_services/Services/source/adcs/adcs_service.c:1271]: If status gets set to failure it probably shouldn't continue execution

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: ex2_services/Services/source/adcs/adcs_service.c (hunk 5 above)
Comment (tganley) [reply ex2_services/Services/source/adcs/adcs_service.c:1271]: Fair

**Review by Taylorr82 — APPROVED** [review]

File: ex2_services/Services/source/adcs/adcs_service.c [hunk 6]
```diff
@@ -1092,42 +1090,54 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
...
+        strncat(file_name, ".txt", 4);
+        int32_t iErr = red_chdir("VOL0:/adcs");
+        int32_t file1 = red_open(file_name, RED_O_RDONLY);
+        if(file1 == -1){
+            uint16_t error = red_errno;
+            sys_log(ERROR, "ADCS_SET_LOG_CONFIG file error %d\r\n", red_errno);
+            status = ADCS_FILE_DNE;
+        }else{
+            char buf[160];
+            red_read(file1, buf, 160);
```
Comment (Taylorr82) [inline ex2_services/Services/source/adcs/adcs_service.c:1109]: Missing check for read failure

File: ex2_services/Services/source/adcs/adcs_service.c [hunk 7]
```diff
@@ -1251,9 +1260,112 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
...
+        int32_t iErr = red_chdir("VOL0:/adcs");
+        if(iErr == -1){
+            sys_log(ERROR, "Error %d trying to change into adcs directory\r\n", red_errno);
+            status = ADCS_FILESYSTEM_FAIL;
+        }else{
+            int32_t file1 = red_open(file_name, RED_O_RDONLY);
+            if(file1 == -1){
+                sys_log(ERROR, "Error %d trying to open file %s\r\n", red_errno, file_name);
+                status = ADCS_FILE_DNE;
+            }else{
```
Comment (Taylorr82) [inline ex2_services/Services/source/adcs/adcs_service.c:1277]: Usually instead of having these big else clauses, the check for failure just returns

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**